internal/analytics/client.go

```diff
@@ -162,7 +160,7 @@ func GetCookieId() uuid.UUID {
 	return id
 }
 
-func Capture(eventType string, providers registry.Providers, data Message, diags diag.Diagnostics, extra ...interface{}) {
+func Capture(eventType string, providers registry.Providers, data Message, extra ...interface{}) {
```

Comment:
Why not keep `diags` param as-is but pass in summarized versions? Repeating the `"diagnostics"` string (and remembering to set `success`) as `extra` is prone to errors (and forgetting). If you don't have any (`sendProviderTelemetryEvents`) could just pass `nil`.

Reply:
Can't pass summarized version as it still would introduce `core` import into the analytics and create an import loop. Moved `SummarizeDiagnostics` into the analytics package.